database/snowflake/snowflake.go

```diff
@@ -117,12 +119,24 @@ func (p *Snowflake) Open(url string) (database.Driver, error) {
 		return nil, ErrNoSchema
 	}
 
+	warehouse := splitPath[3]
+	if len(warehouse) == 0 {
+		return nil, ErrNoWarehouse
+	}
+
+	role := purl.Query().Get("role")
+	if role == "" {
+		return nil, ErrNoRole
+	}
+
 	cfg := &sf.Config{
```

Comment:
It may make sense to:
1. Filter out migrate specific params from the connection URI
1. Rely on [sf.Config.ParseDSN](https://godoc.org/github.com/snowflakedb/gosnowflake#ParseDSN) to parse and validate the connection URI for you
1. Pass the filtered and validated connection URI to `sql.Open()`

Reply:
Done.